src/python/pants/option/optionable.py

```diff
@@ -6,14 +6,31 @@
                         unicode_literals, with_statement)
 
 import re
+from collections import namedtuple
+from hashlib import sha1
 
 from six import string_types
+from twitter.common.collections import OrderedSet
 
+from pants.option.arg_splitter import GLOBAL_SCOPE
 from pants.option.errors import OptionsError
 from pants.option.scope import ScopeInfo
+from pants.util.memo import memoized_method, memoized_property
 from pants.util.meta import AbstractClass
 
 
+class SubsystemDependency(namedtuple('_SubsystemDependency', ('subsystem_cls', 'scope'))):
```

Comment:
If possible, recommend using `pants.util.objects.datatype` instead... `namedtuple` does not count the type of the object in equality, which can lead to weird errors.

Reply:
Done, here and for the `Register` class a few lines below.